components/global/mixer-adapter/cmd/mixer-adapter/main.go

```diff
@@ -65,40 +83,126 @@ func main() {
 		}
 	}
 
-	/* Mutual TLS feature to secure connection between workloads
-	   This is optional. */
-	adapterCertificate := os.Getenv(grpcAdapterCertificatePath) // adapter.crt //change the name
-	adapterprivateKey := os.Getenv(grpcAdapterPrivateKeyPath)   // adapter.key
+	// Mutual TLS feature to secure connection between workloads. This is optional.
+	adapterCertificate := os.Getenv(grpcAdapterCertificatePath) // adapter.crt
+	adapterPrivateKey := os.Getenv(grpcAdapterPrivateKeyPath)   // adapter.key
 	caCertificate := os.Getenv(caCertificatePath)               // ca.pem
+
 	spServerUrl := os.Getenv(spServerUrlPath)
 
+	directory := os.Getenv(directory)
+	waitingSec, _ := strconv.Atoi(os.Getenv(waitingTimeSecEnv))
+	waitingSize, _ := strconv.Atoi(os.Getenv(waitingSizeEnv))
+	tickerSec, _ := strconv.Atoi(os.Getenv(tickerSecEnv))
+
+	vendor := os.Getenv(vendorEnv)
+	dsn := os.Getenv(dsnEnv)
+
+	persist, err = strconv.ParseBool(os.Getenv(shouldPersist)) // this should be a string contains a boolean, "true" or "false"
+	if err != nil {
+		logger.Warnf("Could not convert env of persisting : %s", err.Error())
+		persist = false
+	}
+
 	logger.Infof("Sp server url : %s", spServerUrl)
+	logger.Infof("Should persist : %s", persist)
 
 	client := &http.Client{}
-	publisher := adapter.SPMetricsPublisher{}
 
 	var serverOption grpc.ServerOption = nil
 
 	if adapterCertificate != "" {
-		serverOption, err = getServerTLSOption(adapterCertificate, adapterprivateKey, caCertificate)
+		serverOption, err = getServerTLSOption(adapterCertificate, adapterPrivateKey, caCertificate)
 		if err != nil {
 			logger.Warn("Server option could not be fetched, Connection will not be encrypted")
 		}
 	}
 
-	spAdapter, err := adapter.New(port, logger, client, publisher, serverOption, spServerUrl)
+	buffer := make(chan string, waitingSize*1000)
+	shutdown := make(chan error, 1)
+	spAdapter, err := adapter.New(port, logger, client, serverOption, spServerUrl, buffer, persist)
 	if err != nil {
 		logger.Fatal("unable to start server: ", err.Error())
 	}
-
-	shutdown := make(chan error, 1)
 	go func() {
 		spAdapter.Run(shutdown)
 	}()
+
+	if persist {
+
+		var ps persister.Persister
+		wrt := &writer.Writer{
+			WaitingTimeSec: waitingSec,
+			WaitingSize:    waitingSize,
+			Logger:         logger,
+			Buffer:         buffer,
+			StartTime:      time.Now(),
+		}
+		ticker := time.NewTicker(time.Duration(tickerSec) * time.Second)
+		pub := &publisher.Publisher{
+			Ticker:      ticker,
+			Logger:      logger,
+			SpServerUrl: spServerUrl,
+			HttpClient:  &http.Client{},
+		}
+
+		if (vendor != "") && (dsn != "") {
+
+			db, err := retrier.Retry(10, 2*time.Second, "CONNECT_SQL", func() (db interface{}, err error) {
```

Comment:
Why are we connecting to SQL now? will this connect to SQL regardless of the writer mode? ie, FileSystem or Database?